### PR TITLE
Add function for checking QoS profile compatibility

### PR DESCRIPTION
**Pull request by jacobperron**
Connected to https://github.com/ros2/rmw/pull/299

### CODE REVIEW

File: test_rmw_implementation/test/test_qos_profile_check_compatible.cpp [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  ret = rmw_qos_profile_check_compatible(
+    rmw_qos_profile_parameter_events, rmw_qos_profile_unknown, &compatible, nullptr, 0u);
+  EXPECT_EQ(ret, RMW_RET_INVALID_ARGUMENT);
+  ret = rmw_qos_profile_check_compatible(
+    rmw_qos_profile_services_default, rmw_qos_profile_unknown, &compatible, nullptr, 0u);
+  EXPECT_EQ(ret, RMW_RET_INVALID_ARGUMENT);
+  ret = rmw_qos_profile_check_compatible(
+    rmw_qos_profile_system_default, rmw_qos_profile_unknown, &compatible, nullptr, 0u);
+  EXPECT_EQ(ret, RMW_RET_INVALID_ARGUMENT);
+}
```
Comment (mjeronimo) [inline test_rmw_implementation/test/test_qos_profile_check_compatible.cpp:78]: I assume the purpose of the tests here are to ensure that the function is vectored properly to its implementation and that the full test suite for QoS compatibility is over in rmw_dds_common.

**Review by mjeronimo — APPROVED** [review]

File: test_rmw_implementation/test/test_qos_profile_check_compatible.cpp (hunk 1 above)
Comment (jacobperron) [reply test_rmw_implementation/test/test_qos_profile_check_compatible.cpp:78]: Yes. These tests also are run for every RMW. This is the reason that the tests are not as comprehensive as the tests in `rmw_dds_common`. I don't think we can make many assumptions about QoS compatibility if we consider an non-DDS RMW.

**Review by mjeronimo — APPROVED** [review]

**Review by ivanpauno — APPROVED** [review]

**Comment by jacobperron** [comment]
After a discussion in https://github.com/ros2/rmw_dds_common/pull/45#discussion_r580144067, I had to refactor the tests (e4cf875
). Unknown values no longer result in an error, and the default profiles now cause warnings so we can make any assumptions about them in these tests.

**Review by ivanpauno — APPROVED** [review]

**Review by fujitatomoya — APPROVED** [review]

**Comment by jacobperron** [comment]
CI: https://github.com/ros2/rmw/pull/299#issue-571521564

**Comment by jacobperron** [comment]
@ros-pull-request-builder retest this please

**Comment by jacobperron** [comment]
@ros-pull-request-builder retest this please

**Pull request merged**